test/extensions/filters/listener/original_dst/original_dst_fuzz_test.cc

```diff
@@ -0,0 +1,69 @@
+#include "envoy/extensions/filters/listener/original_dst/v3/original_dst.pb.h"
+
+#include "common/network/utility.h"
+
+#include "extensions/filters/listener/original_dst/original_dst.h"
+
+#include "test/extensions/filters/listener/original_dst/original_dst_fuzz_test.pb.validate.h"
+#include "test/fuzz/fuzz_runner.h"
+#include "test/mocks/network/mocks.h"
+
+#include "gmock/gmock.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace ListenerFilters {
+namespace OriginalDst {
+
+DEFINE_PROTO_FUZZER(
+    const envoy::extensions::filters::listener::original_dst::v3::OriginalDstTestCase& input) {
+
+  try {
+    TestUtility::validate(input);
+  } catch (const ProtoValidationException& e) {
+    ENVOY_LOG_MISC(debug, "ProtoValidationException: {}", e.what());
+    return;
+  }
+
+  NiceMock<Network::MockListenerFilterCallbacks> callbacks_;
+  NiceMock<Network::MockConnectionSocket> socket_;
```

Comment:
Ran it locally and got around ~240 exec/s - anything I can do to make it faster?

Reply:
The googlemock library isn't designed for performance. You could probably write your own subclass of ConnectionSocket, (or subclass of the mock class) that would execute the configured actions without all the mock stuff in between without too much difficulty.

Reply:
+1 Specially if it would also be a re-useable class component in most listener filter fuzzers.
Is it possible to use a fake (rather than a mock) or make it as real as possible?

Reply:
I'm now getting 4096 exec/s after switching to my own subclass of MockConnectionSocket in the latest commit (could eventually use a subclass of ConnectionSocket) - coverage is unchanged.